### PR TITLE
Eliminated two possible false positives from BlockingObservableTest

### PR DESCRIPTION
**Pull request by illusioni**

### CODE REVIEW

**Comment by zsxwing** [comment]
Could you clarify why you need to change them, such as provide an example of `false positives`?

**Comment by illusioni** [comment]
Hi Shixiong Zhu,
The existing test is wrong because it does not test all that's specified.
It only works because the component under test (CUT) behaves well.
In the case of the "toFuture function throwing rather than producing a failing future" the existing test would still succeed (that's the false positive).
My version of the test would fail - as a test should when the CUT does not comply with the spec.
Hope that info helps. :-)
Big fan of Rx by now...
Klaus
On Oct 6, 2015 17:00, Shixiong Zhu notifications@github.com wrote:Could you clarify why you need to change them, such as provide an example of false positives?

File: src/test/scala/rx/lang/scala/observables/BlockingObservableTest.scala [hunk 1]
```diff
@@ -161,15 +161,23 @@ class BlockingObservableTest extends JUnitSuite {
...
 
-  @Test(expected = classOf[IllegalArgumentException])
+  @Test
   def testToFutureWithMultipleItems() {
     val o = Observable.just(1, 2)
-    Await.result(o.toBlocking.toFuture, 10 seconds)
+    val future = o.toBlocking.toFuture //if this were to throw the original test would wrongly succeed
+    Await.result(future.failed, 10 seconds) match {
+      case t:IllegalArgumentException => //this is what we expect
+      case _ => fail("expected the future to fail with a NoSuchElementException")
```
Comment (zsxwing) [inline src/test/scala/rx/lang/scala/observables/BlockingObservableTest.scala]: nit: NoSuchElementException -> IllegalArgumentException

**Comment by zsxwing** [comment]
@illusioni Thanks for your clarification. I will merge this PR once you fix the typo.

**Comment by illusioni** [comment]
I tried hard, @zsxwing, but I can't find a typo in my changes.
Please let me know where you see an error

**Comment by illusioni** [comment]
Aaah. Is this what you meant?

**Comment by zsxwing** [comment]
@illusioni Exactly. Thanks! Merged.

**Pull request merged**